lib/KImg.vue

```diff
@@ -0,0 +1,159 @@
+      :alt="alternateText"
+      :style="cssVars"
+    >
+    <slot></slot>
+  </div>
+
+</template>
+
+
+<script>
+
+  export default {
+    name: 'KImg',
+    props: {
+      /**
+       * Sets the image path
+       */
+      src: {
+        type: String,
+        default: null,
+      },
+      /**
+       * Alternate text for the image. This is required and will throw a warning when it's not provided (empty) unless isDecorative is true
+       */
+      altText: {
+        type: String,
+        default: '',
+      },
+      /**
+       * Sets the image as decorative. This sets the alt image property to an empty string.
+       */
+      isDecorative: {
+        type: Boolean,
+        default: false,
+      },
+      /**
+       * Sets the height for the component
+       */
+      height: {
+        type: [Number, String],
+        default: undefined,
+      },
+      /**
+       * Sets the width for the component
+       */
+      width: {
+        type: [Number, String],
+        default: undefined,
+      },
+      /**
+       * Sets the maximum height for the component
+       */
+      maxHeight: {
+        type: [Number, String],
+        default: undefined,
+      },
+      /**
+       * Sets the minimum height for the component
+       */
+      minHeight: {
+        type: [Number, String],
+        default: undefined,
+      },
+      /**
+       * Sets the maximum width for the component
+       */
+      maxWidth: {
+        type: [Number, String],
+        default: undefined,
+      },
+      /**
+       * 	Sets the minimum width for the component
+       */
+      minWidth: {
+        type: [Number, String],
+        default: undefined,
+      },
+    },
+    computed: {
+      alternateText() {
+        return this.isDecorative ? '' : this.altText;
+      },
+      imgHeight() {
+        return this.validateAndFormatUnits(this.height);
+      },
+      imgWidth() {
+        return this.validateAndFormatUnits(this.width);
+      },
+      imgMaxHeight() {
+        return this.validateAndFormatUnits(this.maxHeight);
+      },
+      imgMinHeight() {
+        return this.validateAndFormatUnits(this.minHeight);
+      },
+      imgMaxWidth() {
+        return this.validateAndFormatUnits(this.maxWidth);
+      },
+      imgMinWidth() {
+        return this.validateAndFormatUnits(this.minWidth);
+      },
+      cssVars() {
+        return {
+          '--height': this.imgHeight,
+          '--width': this.imgWidth,
+          '--max-height': this.imgMaxHeight,
+          '--min-height': this.imgMinHeight,
+          '--max-width': this.imgMaxWidth,
+          '--min-width': this.imgMinWidth,
+        };
+      },
+    },
+    created() {
+      if (!this.isDecorative && !this.altText) {
+        throw new Error('Missing required prop - provide altText or indicate isDecorative');
+      }
+    },
+    methods: {
+      validateAndFormatUnits(propValue) {
+        if (propValue) {
+          if (Number.isInteger(propValue)) {
```

Comment:
This check would fail for a `propValue` that is a floating point number(eg 30.5).  The same `propValue` but with a unit(eg 30.5rem) would pass the validation. This not only causes an inconsistency that needs to be streamlined but also means that the regex fails for floating point numbers with units(eg 30.5rem). Maybe using the `isNaN` is a better option. This would also mean updating the regex to capture floating point numbers as well. I tweaked your regex to cater for the decimal points part and the final result is this `(\d*\.\d+)([\s\S]*)`

Reply:
This is another update to my earlier recommended regex. using `(\d*\.?\d+)([a-zA-Z,\s]*)`  instead will ensure that values like 30px, 12.5rem would pass validation. However values like 10.em would fail validation as such an input would not work when applied in css.